packages/create-sitecore-jss/src/templates/nextjs/src/pages/api/revalidate.ts

```diff
@@ -0,0 +1,38 @@
+﻿import { RevalidateMiddleware } from '@sitecore-jss/sitecore-jss-nextjs/revalidate';
+import { NextApiResponse, NextApiRequest } from 'next';
+import clientFactory from 'lib/graphql-client-factory';
+
+/**
+ * Nextjs API route /api/revalidate
+ * This is a revalidate endpoint that can be used to trigger Incremental Static Regeneration (ISR).
+ * of Next.js pages on-Demand.This endpoint is called by Sitecore when content is published,
+ * and can also be called manually to trigger revalidation of pages.
+ * @param req  The Next.js API request object
+ * @param res  The Next.js API response object
+ */
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { query, body } = req;
+  const secret = query['secret'];
+  const revalidateHandler = new RevalidateMiddleware({
+    clientFactory,
+    multiSite: body.multiSite || false, // override this value through your webhook payload if using multi-site add-on
+    personalize: body.personalize || false, // override this value through your webhook payload if personalization is configured
+  }).getHandler();
+
+  /**
+   NOTE: It is highly recommended to add a secret to your revalidate endpoint before going to production.
+   This prevents unauthorized users from triggering revalidation requests.
+   */
+  if (process.env.ISR_REVALIDATE_SECRET) {
+    if (secret && secret !== process.env.ISR_REVALIDATE_SECRET) {
```

Comment:
Should also be considered a 401 if the secret is missing
```suggestion
    if (!secret || secret !== process.env.ISR_REVALIDATE_SECRET) {
```